### DISCUSSION TITLE
pnpm update -S to write actual version to every dependencies

### DISCUSSION THREAD

**Discussion by u007** [pnpm/pnpm#16177] [Ideas]
would like to see how this work, inspired by how ```npm update -S``` works.

**Describe the user story**

currently pnpm update works, but it does not write the version to package.json

**Describe the solution you'd like**

lets say my package.json have dependencies of:
```
"aws-sdk": "*",
"bcrypt": "*",
```
would like to see it turn to this after pnpm update -S

```
"aws-sdk": "^2.1227.0",
"bcrypt": "^5.0.1",
```

**Comment by zkochan** [comment]
Sounds good.

**Comment by LuciNyan** [comment]
Hi! I'd love to work on this issue! Could you please assign this to me if not been assigned to anybody yet? Thanks!

**Comment by LuciNyan** [comment]
Hi! @zkochan. I have a little confused. When using npm update, if we don't add `-S`, package.json will actually not be modified. But when using pnpm update, package.json is modified. What I want to confirm is whether we need to add a `-S` parameter to `pnpm update` or just change the behaviour of the current `update`.

**Comment by zkochan** [comment]
just change the behaviour of the current update.

**Comment by karlhorky** [comment]
@zkochan so this will allow for updating to an exact version of a dependency too?

Eg. switching from `typescript@^6.0.3` to `typescript@6.0.3` (without `^`) as in this comment:

- https://github.com/pnpm/pnpm/issues/6398#issuecomment-5314388945

Otherwise issue #6398 maybe shouldn't have been [closed as a duplicate](https://github.com/pnpm/pnpm/issues/6398#issuecomment-5745808283).
